repo2docker/buildpacks/nix/nix-shell-wrapper

```diff
@@ -8,8 +8,13 @@ _term() {
 
 trap _term SIGTERM
 
-echo "$*"
-nix-shell default.nix --command "$*" &
+echo 'command' "$*"
+
+if [ -f ./start ] then
+   nix-shell default.nix --command ./start "$*" &
```

Comment:
Need to be something like `nix-shell default.nix --command "./start && $*" &`. This is why I have to write a wrapper in the first place. It only looks at the first argument.

Reply:
Struggling to understand/make this work :-/ What I was trying to achieve (and then had to leave in this half done state for a call) is to pass the arguments on to `start` because `start` scripts [expect to be given arguments that they should `exec` when they are done](https://repo2docker.readthedocs.io/en/latest/config_files.html#start-run-code-before-the-user-sessions-starts).

Reply:
Okay so now I understand. How does `nix-shell default.nix --command "./start $*" &` work?